ladybug/designday.py

```diff
@@ -0,0 +1,1109 @@
+from .location import Location
+from .futil import write_to_file
+from .dt import DateTime
+from .header import Header
+from .datacollection import DataCollection
+from .datatype import DataPoint
+from .analysisperiod import AnalysisPeriod
+from .sunpath import Sunpath
+
+from .skymodel import ashrae_revised_clear_sky
+from .skymodel import ashrae_clear_sky
+from .skymodel import energyplus_horizontal_infrared
+
+from .psychrometrics import rel_humid_from_db_dpt
+from .psychrometrics import dew_point_from_db_hr
+from .psychrometrics import dew_point_from_db_enth
+from .psychrometrics import dew_point_from_db_wb
+
+import math
+import os
+import re
+import codecs
+import platform
+
+day_types = ['SummerDesignDay', 'WinterDesignDay', 'Sunday', 'Monday',
+             'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Holiday',
+             'CustomDay1', 'CustomDay2']
+h_types = ['Wetbulb', 'Dewpoint', 'HumidityRatio', 'Enthalpy']
+sky_types = ['ASHRAEClearSky', 'ASHRAETau', 'ZhangHuang', 'Schedule']
+comments = ['!- Name',
+            '!- Month',
+            '!- Day of Month',
+            '!- Day Type',
+            '!- Max Dry-Bulb Temp [C]',
+            '!- Daily Dry-Bulb Temp Range [C]',
+            '!- Dry-Bulb Temp Range Modifier Type',
+            '!- Dry-Bulb Temp Range Modifier Schedule Name',
+            '!- Humidity Condition Type',
+            '!- Wetbulb/Dewpoint at Max Dry-Bulb [C]',
+            '!- Humidity Indicating Day Schedule Name',
+            '!- Humidity Ratio at Maximum Dry-Bulb {kgWater/kgDryAir}',
+            '!- Enthalpy at Maximum Dry-Bulb {J/kg}',
+            '!- Daily Wet-Bulb Temperature Range [deltaC]',
+            '!- Barometric Pressure [Pa]',
+            '!- Wind Speed {m/s}',
+            '!- Wind Direction [Degrees; N=0, S=180]',
+            '!- Rain [Yes/No]',
+            '!- Snow on ground [Yes/No]',
+            '!- Daylight Savings Time Indicator',
+            '!- Solar Model Indicator',
+            '!- Beam Solar Day Schedule Name',
+            '!- Diffuse Solar Day Schedule Name',
+            '!- ASHRAE Clear Sky Optical Depth for Beam Irradiance (taub)',
+            '!- ASHRAE Clear Sky Optical Depth for Diffuse Irradiance (taud)',
+            '!- Clearness [0.0 to 1.2]']
+temp_multipliers = [0.82, 0.88, 0.92, 0.95, 0.98, 1, 0.98, 0.91, 0.74,
+                    0.55, 0.38, 0.23, 0.13, 0.05, 0, 0, 0.06, 0.14, 0.24,
+                    0.39, 0.5, 0.59, 0.68, 0.75]
+
+
+class Ddy(object):
+    """Import data from a local .ddy file.
+
+    properties:
+        location: A Ladybug location object
+        design_days: A list of the design days in the ddy file.
+        file_path: Optional file path into which ddy file can be written.
+    """
+    # TODO: set the default file path to use the ladybug default folder.
+    def __init__(self, location, design_days=[], file_path=None):
```

Comment:
Doesn't design_days already include location data? Based on `from_design_day` it does. Then the question will be should we remove location input and get it from design_days?

Also I'm not sure what does a DDY class with no design_days mean? Should we remove empty list as the default input?

Reply:
I spent a long time thinking about this.

The issue is that a ddy file can have only one location that is the same for all of the design days in the file.
However each design day has its own location, which is necessary for all of the solar methods that are on the design day object.
This creates the possibility in some workflows of someone creating a ddy file where all of the design days have different locations.  Therefore, we need a location property on the Ddy object that synchronizes all of the design days connected to the .ddy with a single location.

Yes, we can remove the default of creating a ddy file with no design days inside of it.

Reply:
This has been addressed in our discussion